src/library.js

```diff
@@ -989,7 +989,12 @@ LibraryManager.library = {
       info.refcount--;
       if (info.refcount === 0) {
         if (info.destructor) {
+#if BINARYEN == 0
```

Comment:
This isn't quite right - binaryen mode means either asm2wasm or the wasm backend. The difference is only in the latter. So we should check `WASM_BACKEND` here instead